perfkitbenchmarker/vpn_service.py

```diff
@@ -0,0 +1,326 @@
+                    'If set, use this name for VPN Service.')
+flags.DEFINE_string('vpn_service_shared_key', None,
+                    'If set, use this PSK for VPNs.')
+flags.DEFINE_string('vpn_service_routing_type', None,
+                    'static or dynamic(BGP)')
+flags.DEFINE_integer('vpn_service_ike_version', None, 'IKE version')
+
+FLAGS = flags.FLAGS
+
+
+def GetVPNServiceClass():
+  """Gets the VPNService class.
+
+  Args:
+
+  Returns:
+    Implementation class
+  """
+  return resource.GetResourceClass(VPNService)
+
+
+class VPN(object):
+  """An object representing the VPN.
+
+  A VPN instance manages tunnel configurations for exactly 1 pair of endpoints.
+  """
+
+  def __init__(self, *args, **kwargs):
+      return object.__init__(self, *args, **kwargs)
+
+  def getKeyFromGWPair(self, gateway_pair, suffix=''):
+    """Return the VPN key for a pair of endpoints.
+
+    Args:
+      gateway_pair: A tuple of 2 VPN gateways which define the VPN tunnel.
+      suffix: A unique suffix if multiple tunnels b/t this gateway pair exist.
+
+    Returns:
+      string. The VPN key.
+
+    """
+    key = 'vpn' + ''.join(gw for gw in gateway_pair) + suffix + FLAGS.run_uri
+    return key
+
+  def Create(self, gateway_pair, suffix=''):
+    self.GWPair = gateway_pair
+    self.name = self.getKeyFromGWPair(gateway_pair)
+    self.tunnel_config = TunnelConfig(tunnel_name=self.name, suffix=suffix)
+
+  def Delete(self):
+    pass
+
+  def GetVPN(self, gateway_pair, suffix=''):
+    """Gets a VPN object for the gateway_pair or creates one if none exists
+
+    Args:
+    gateway_pair: a tuple of two VPNGWs
+    """
+
+    benchmark_spec = context.GetThreadBenchmarkSpec()
+    if benchmark_spec is None:
+      raise errors.Error('GetVPN called in a thread without a '
+                         'BenchmarkSpec.')
+    with benchmark_spec.vpns_lock:
+      key = self.getKeyFromGWPair(gateway_pair, suffix)
+      if key not in benchmark_spec.vpns:
+        self.Create(gateway_pair, suffix)
+        benchmark_spec.vpns[key] = self
+      return benchmark_spec.vpns[key]
+
+  def ConfigureTunnel(self):
+    """Configure the VPN tunnel."""
+
+    benchmark_spec = context.GetThreadBenchmarkSpec()
+    vpn_gateway_0 = benchmark_spec.vpn_gateways[self.GWPair[0]]
+    vpn_gateway_1 = benchmark_spec.vpn_gateways[self.GWPair[1]]
+
+    assert not (vpn_gateway_0.require_target_to_init and vpn_gateway_1.require_target_to_init), 'Cant connect 2 passive VPN GWs'
+
+    tunnel_config_hash = None
+    while not self.isTunnelConfigured():
+        vpn_gateway_0.ConfigureTunnel(self.tunnel_config)
+        vpn_gateway_1.ConfigureTunnel(self.tunnel_config)
+        if self.tunnel_config.hash() == tunnel_config_hash:
+          raise ValueError('Not enough info to configure tunnel.')
+        tunnel_config_hash = self.tunnel_config.hash()
+
+    tunnel_status = self.isTunnelReady()
+    logging.info('Tunnel is ready?: %s ' % tunnel_status)
```

Comment:
logging.debug('Tunnel is ready?: %s ', tunnel_status)

Reply:
Changed most logging to debug since it was getting noisy.